Doc update for wifi credentials library and EAP methods (#23022)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Wi-Fi®
------

* Added:

>   * Enterprise security now supports EAP-PEAP and EAP-TTLS authentication methods in Wi-Fi management.

Reviewer: Shouldn't we also have an entry for the Wi-Fi credentials library removal in the release notes?
If the code and doc both are removed in this release, then in the lib section, else in the doc section.

>   * Enterprise security now supports EAP-PEAP and EAP-TTLS authentication methods in Wi-Fi management.

Author: Ok sure, Wi-Fi credentials library was moved upstream earlier. This change is regarding moving document of Wi-Fi credentials library upstream.
I will make entry in doc section.

> * Added:
> 
>   * Added support for EAP-PEAP and EAP-TTLS authentication methods to enterprise security in the Wi-Fi management API.

Reviewer 2: ```suggestion
* Added support for EAP-PEAP and EAP-TTLS authentication methods to enterprise security in the Wi-Fi management API.
```

* Added the :ref:`log_rpc` library documentation page.
* Removed the Getting started with nRF7002 DK and Getting started with other DKs pages from the :ref:`gsg_guides` section.
  These pages were no longer relevant as the `Quick Start app`_ now also supports the nRF7002 DK.

> * The Wi-Fi credentials library documentation has been moved to the upstream zephyr.

Reviewer 2: Can we add a link to the doc here? Also, move it before the previous bullet "Removed the Getting started.....".
```suggestion
* Moved the Wi-Fi credentials library page to the upstream Zephyr repository.
```
